vm/src/stdlib/collections.rs

```diff
@@ -591,6 +590,19 @@ mod _collections {
             }
         }
 
+        #[pyslot]
+        fn tp_new(cls: PyTypeRef, args: FuncArgs, vm: &VirtualMachine) -> PyResult<PyRef<Self>> {
+            if args.args.len() <= 1 {
+                let deque = args.bind(vm)?;
+                PyDequeIterator::new(deque).into_ref_with_type(vm, cls)
+            } else {
+                let (deque, position) = args.bind(vm)?;
```

Comment:
`index` (i.e `position` here) should be allowed as a positional and a keyword. Right now it's only positional. Other than that, it lgtm.

Reply:
Hm, in addition to that, you'd need to check that the index passed in isn't negative and that it isn't `> deque.len`. On first thought, in the first case you should silently ignore it and in the second you should set the iterator to exhausted.

Reply:
> `index` (i.e `position` here) should be allowed as a positional and a keyword. Right now it's only positional. Other than that, it lgtm.

can you give more information about what "positional" and "keyword" means?

Reply:
Essentially, the iterators should be able to be called in three ways:

```
rev_it = type(reversed(deque()))
rev_it(deque())                 # only deque specified
rev_it(deque(), 1)             # deque and index as a positional arg specified
rev_it(deque(), index=1)  # deque and index as a keyword arg specified
```

In short, it should be easier to specify `IterNewArgs` (see `PyDequeOptions` for insipation) with `index` being `optional`.

Reply:
Ah, i got it, thanks! i will send v2 code till tomorrow

Reply:
@DimitrisJim  I checked the keyword case using the code
```
from collections import deque

rev_it = type(reversed(deque()))
c = rev_it(deque('abcd'), index=1)

for i in c:
    print(i)
```
and the index key doesn't work. it prints d,c,b,a
also, in [CPython](https://github.com/python/cpython/blob/eb263f9a356f5c5f21b8d5ce20bac92f31c40cad/Modules/_collectionsmodule.c#L1775) it is using ParseTuple in `dequeiter_new`. So, I think only positional parameter is allowed in `__new__`

I added extra_test in snippet. in CPython 3.8, the test code works well also with the commented test. however, in RustPython, it detects the wrong keyword index and return error. So, I commented the keyword test in extra_test and remained it as TODO

Reply:
Add an unused parameter with type `rustpython_vm::function::KwArgs` like `_kwargs: KwArgs`. Then it will take any keyword arguments but ignore them

Reply:
I fixed with v3 code. Thank you for the advice!

Reply:
Yup, it actually allows any number of kwargs but ignores them as you say. Can't see how this is the intended behavior, an issue about it should probably be opened.